docs/inference.md

```diff
@@ -0,0 +1,147 @@
+
+### Example
+
+You need a [trained model](../wmt/README.md).
+
+You need to be careful to apply the same preprocessing to your test data that you applied at training time, including
+any [subword processing](http://github.com/rsennrich/subword-nmt), since Sockeye itself does not do this.
+
+Constraints must be encoded with a JSON object.
+This JSON object can be produced with the provided script:
+
+    echo -e "This is a test .\tconstraint\tmulti@@ word const@@ raint" \
+      | python3 -m sockeye.lexical_constraints
+
+The script creates a Python object with the constraints encoded as follows:
+
+    { 'text': 'This is a test .', 'constraints': ['constr@@ aint', 'multi@@ word constr@@ aint'] }]
+
+You can pass the output of this to Sockeye.
+Make sure that you (a) the JSON object is on a *single line* and (b) you pass `--json-input` to Sockeye, so that it knows to parse the input (without that flag, it will treat the JSON input as a regular sentence).
+We also recommend that you increase the beam a little bit and enable beam pruning:
+
+    echo -e "This is a test .\tconstraint\tmultiword constraint" \
+      | python3 -m sockeye.lexical_constraints \
+      | python3 -m sockeye.translate -m /path/to/model --json-input --beam-size 20 --beam-prune 20 [other args]
+
+You will get a translation with the required constraints as part of the output.
+
+### Negative constraints
+
+Negative constraints---phrases that must *not* appear in the output---are also supported.
+To use them, use the "avoid" key in the JSON object instead of "constraints".
+An example JSON object:
+
+    { 'text': 'This is a test .', 'avoid': ['Test'] }
+
+Multiple negative constraints and multi-word negative constraints are supported, just like for positive constraints.
+You can also add `<s>` and `</s>` to constraints, to specify that phrases that must not start or end a sentence.
+Don't forget to apply your preprocessing!
+
+### Scoring
+
+Lexical constraints can also be used as a rudimentary scoring mechanism, by providing the entire reference (with `<s>` and `</s>` tokens) as a constraint.
+(However, you'll get significantly faster results with [Sockeye's scoring module](scoring.html)).
+For example:
+
+    echo '{ "text": "This is a test", "constraints": ["<s> Dies ist ein Test </s>"] }' \
+      python3 -m sockeye.translate -m /path/to/model --json-input --beam-size 1 --output-type translation_with_score
+
+This will output tab-delimited pairs of (score, translation).
+As always, don't forget to apply source- and target-side preprocessing to your input and your constraint.
+
+## CPU process per core translation
+
+On multi-core computers, translation per core separately can speedup translation performance, due to some operation can't be handled parallel in one process.
+Using this method, translation on each core can be parallel.
+
+One [python script example](tutorials/cpu_process_per_core_translation.py] is given and you can run it as follows:
```

Comment:
there's a typo in the link (final parentheses). Also: would it actually work to point to a file in the repo this way?

Reply:
Fixed. Does this even work?

Reply:
no, I think it was just displayed as plain text.

Reply:
I mean, does the multi-core CPU processing still work?

Reply:
ahh :) probably, but not sure. I would have probably left this in the tutorials and not put it under the inference page, as it might create more confusion to users.